pkg/runtime/runtime.go

```diff
@@ -2150,6 +2170,17 @@ func (a *DaprRuntime) isComponentAuthorized(component componentsV1alpha1.Compone
 	return true
 }
 
+func (a *DaprRuntime) IsComponentLoaded(component componentsV1alpha1.Component) bool {
+	a.componentsLock.RLock()
+	defer a.componentsLock.RUnlock()
+	for _, loadedComp := range a.components {
+		if reflect.DeepEqual(loadedComp, component) {
```

Comment:
Is there a better way to check if a component is already loaded besides using `reflect.DeepEqual`? Since the properties of `component` can change throughout its lifecycle, I presume?

Reply:
Updated to compare using fields that are read only throughout component lifecycle (ObjectMetadata.Name, Spec.Type, Version, TypeMeta)
https://pkg.go.dev/k8s.io/apimachinery@v0.23.0/pkg/apis/meta/v1?utm_source=gopls#ObjectMeta